pallets/parachain-staking/src/test_reward_payout.rs

```diff
@@ -146,9 +150,182 @@ fn end_to_end_happy_path() {
+
+    let collator1_points_percentage =
+        Perbill::from_rational(COLLATOR1_POINTS, TOTAL_POINTS_FOR_ERA);
+    let collator1_total_reward = collator1_points_percentage * expected_total_reward;
+
+    let collator2_points_percentage =
+        Perbill::from_rational(COLLATOR2_POINTS, TOTAL_POINTS_FOR_ERA);
+    let collator2_total_reward = collator2_points_percentage * expected_total_reward;
+
+    return (collator1_total_reward, collator2_total_reward)
+}
+
+mod compute_total_reward_to_pay {
+    use super::*;
+
+    #[test]
+    fn works_as_expected() {
+        let reward_pot_account_id = ParachainStaking::compute_reward_pot_account_id();
+        ExtBuilder::default()
+            .with_balances(vec![(COLLATOR1, 10000), (COLLATOR2, 10000), (TX_SENDER, 10000)])
+            .with_candidates(vec![
+                (COLLATOR1, COLLATOR1_OWN_STAKE),
+                (COLLATOR2, COLLATOR2_OWN_STAKE),
+            ])
+            .build()
+            .execute_with(|| {
+                roll_to_era_begin(2);
+
+                // Initial state
+                assert_eq!(ParachainStaking::locked_era_payout(), 0);
+
+                let expected_total_reward = expected_tx_fee() + TIP;
+                let (collator1_reward, collator2_reward) =
+                    set_reward_pot_and_trigger_payout(ERA_BLOCKS_HAVE_BEEN_AUTHORED, 3);
+
+                // Reward pot balance has decreased
+                assert_eq!(
+                    Balances::free_balance(&reward_pot_account_id),
+                    expected_total_reward - collator1_reward
+                );
+
+                // Locked era payout has reserved enough to pay remaining collators
+                assert_eq!(ParachainStaking::locked_era_payout(), collator2_reward);
+
+                // Send another transaction and generate income
+                pay_gas_for_transaction(&TX_SENDER, TIP * 2);
+
+                // Show that reward pot balance has increased due to the new transaction paying a
+                // fee
+                assert_eq!(
+                    Balances::free_balance(&reward_pot_account_id),
+                    collator2_reward + expected_tx_fee() + TIP * 2
+                );
+
+                // Locked era payout did not change
+                assert_eq!(ParachainStaking::locked_era_payout(), collator2_reward);
+
+                // Compute new reward to distribute. This will lock the new amount.
+                let new_total_reward_to_pay = ParachainStaking::compute_total_reward_to_pay();
+
+                // Locked era payout has increased
+                assert_eq!(
+                    ParachainStaking::locked_era_payout(),
+                    collator2_reward + expected_tx_fee() + TIP * 2
+                );
+
+                // We can only pay out new income. collator2_reward has already been locked for
+                // payment
+                assert_eq!(new_total_reward_to_pay, expected_tx_fee() + TIP * 2);
```

Comment:
Shall we just "complete the test" by paying out collator2's reward and see that what remains is the fee and tip of the last transaction?

Reply:
This has already been covered by the happy path test above. And Line 229 shows that we have enough to cover the payout for collator2.